packages/features/ee/api-keys/lib/apiKeys.ts

```diff
@@ -8,3 +9,20 @@ export const generateUniqueAPIKey = (apiKey = randomBytes(16).toString("hex")) =
   hashAPIKey(apiKey),
   apiKey,
 ];
+
+export const apiKeySchema = z.string({ required_error: "No apiKey provided" }).refine(
+  (value) => {
+    // Check if it starts with process.env.API_KEY_PREFIX
+    if (!value.startsWith(process.env.API_KEY_PREFIX || "cal_")) {
+      return false;
+    }
+    // Extract the hash part and validate its format
+    const hashPart = value.slice(4); // Remove prefix
```

Comment:
Can `process.env.API_KEY_PREFIX` not be other than 4 in length? If it can, this might fail no?